python/love/commander/reports.py

```diff
@@ -162,9 +163,15 @@ def act(index, actuator) -> int:
     reports_app.router.add_post("/m1m3-bump-tests", query_m1m3_bump_tests)
     reports_app.router.add_post("/m1m3-bump-tests/", query_m1m3_bump_tests)
 
-    async def on_cleanup(reports_app):
-        # This app doesn't require cleaning up.
-        pass
+    async def on_cleanup(app):
+        global efd_clients
+        for client in efd_clients:
```

Comment:
Better - see above:
```
        for index, client in efd_clients.items():
            if client is not None:
                try:
                    client.close()
                except Exception as e:
                    logging.error(f"Error closing EFD client {index}: {e}")
        efd_clients = dict()```
        
  Also with EfdClient as class, reports can be made subclass and call that on its own. You can then also use desctructor (__done__? ) instead of on_cleanup.

Reply:
Thanks @pkubanek ! I like that idea, I'll create a ticket for giving it a shot eventually :+1: .